File: 232 - Implement Queue using Stacks/cpp/ans.cpp
class MyQueue {
public:
    stack<int> s;
    int firstValue;
    
    MyQueue() {
        
    }
    
    void push(int x) {
        if (s.empty()) {
            firstValue = x;
        }
        s.push(x);

        return;
    }
    
    int pop() {
        stack<int> temp;
        // save the value to temp stack
        // and exclude the first data
        for(int i = s.size(); i > 1; i--){
            temp.push(s.top());
            s.pop();
        }
        // pop the first data
        int ret = s.top();
        s.pop();
        
        while (!temp.empty()) {
            int x = temp.top();
            if (s.empty()) {
                firstValue = x;
            }
            s.push(x);
            temp.pop();
        }
        
        return ret;
    }
    
    int peek() {
        return firstValue;
    }
    
    bool empty() {
        return s.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: 232 - Implement Queue using Stacks/python/ans.py
class MyQueue:

    def __init__(self):
        self.array = []
        self.first_value = -1

    def push(self, x: int) -> None:
        if len(self.array) == 0:
            self.first_value = x
        self.array.append(x)

    def pop(self) -> int:
        temp = list()
        while len(self.array) > 0:
            x = self.array[-1]
            self.array = self.array[:-1]
            temp.append(x)
        ret = temp[-1]
        temp = temp[:-1]
        while len(temp) > 0:
            x = temp[-1]
            temp = temp[:-1]
            if len(self.array) == 0:
                self.first_value = x
            self.array.append(x)
        return ret

    def peek(self) -> int:
        return self.first_value

    def empty(self) -> bool:
        return len(self.array) == 0


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()